packages/@uppy/drop-target/src/index.ts

```diff
@@ -1,24 +1,47 @@
+import type { Body, Meta, UppyFile } from '@uppy/utils/lib/UppyFile'
+import type { Uppy, State } from '@uppy/core/src/Uppy.ts'
+import type { UIPluginOptions } from '@uppy/core/src/UIPlugin.ts'
 import BasePlugin from '@uppy/core/lib/BasePlugin.js'
 import getDroppedFiles from '@uppy/utils/lib/getDroppedFiles'
 import toArray from '@uppy/utils/lib/toArray'
-
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// @ts-ignore We don't want TS to generate types for the package.json
 import packageJson from '../package.json'
 
-function isFileTransfer (event) {
-  return event.dataTransfer.types?.some((type) => type === 'Files') ?? false
+interface DropTargetOptions extends UIPluginOptions {
+  target?: HTMLElement | string | null
+  onDrop?: (event: DragEvent) => void
+  onDragOver?: (event: DragEvent) => void
+  onDragLeave?: (event: DragEvent) => void
+}
+
+interface DragEventWithFileTransfer extends DragEvent {
+  dataTransfer: NonNullable<DragEvent['dataTransfer']>
+}
+
+function isFileTransfer(event: DragEvent): event is DragEventWithFileTransfer {
+  return event.dataTransfer?.types?.some((type) => type === 'Files') ?? false
 }
 
 /**
  * Drop Target plugin
  *
  */
-export default class DropTarget extends BasePlugin {
+export default class DropTarget<
+  M extends Meta,
+  B extends Body,
+> extends BasePlugin<DropTargetOptions, M, B> {
```

Comment:
Can you please refactor this to use the `DefinePluginOpts` that was introduced in https://github.com/transloadit/uppy/pull/4885?
```suggestion
> extends BasePlugin<DefinePluginOpts<DropTargetOptions, keyof typeof defaultOptions>, M, B> {
```